chia/pools/pool_wallet.py

```diff
@@ -448,14 +449,24 @@ async def create_new_pool_wallet_transaction(
         )
         return standard_wallet_record, p2_singleton_puzzle_hash, launcher_coin_id
 
-    async def sign(self, coin_spend: CoinSpend) -> SpendBundle:
+    async def sign(self, coin_spend: CoinSpend, fee: uint64) -> SpendBundle:
         async def pk_to_sk(pk: G1Element) -> PrivateKey:
             owner_sk: Optional[PrivateKey] = await find_owner_sk([self.wallet_state_manager.private_key], pk)
-            assert owner_sk is not None
+            # assert owner_sk is not None
+            if owner_sk is None:
+                return self.standard_wallet.secret_key_store.secret_key_for_public_key(pk)
             return owner_sk
 
+        spends = [coin_spend]
+        if fee > 0:
+            # TODO: test failure UX when fee is higher than available balance
+            get_max_send_amount = await self.standard_wallet.get_max_send_amount()
+            assert get_max_send_amount > fee
+            my_address = await self.standard_wallet.get_new_puzzlehash()
+            fee_spends = await self.standard_wallet._generate_unsigned_transaction(uint64(0), my_address, fee)
```

Comment:
We need to assert an announcement when creating fee spends so that a malicious farmer can't separate the fee from the thing being paid for.  I'm not sure why a function called `sign` should have fee logic inside of it?  Shouldn't you just be passing spends to this function and having it sign with the PKs and msgs it detects?

Reply:
Yes, this is exactly the main reason I wanted you and Richard to review this - the code that follows attempts to join the two spends. But Richard points out the same improvement - it would be better if the fee were includedd in the initial, unsigned spend.

The code that creates this CoinSpend is in the driver code `chia/pools/pool_puzzles.py:create_travel_spend` - could Quex or Richard add a fee parameter here, please?

Reply:
You can use `generate_signed_transaction` for this fee spend can't you?  Then you can remove this fee parameter from this function that really shouldn't know anything about fees or creating transactions.  Once both the spend and the fee spend are signed, you can aggregate the bundle together and also importantly push both tx records rather than just forgetting about the fee one that you generated in here.